internal/gw/publish.go

```diff
@@ -64,7 +64,13 @@ func (c *client) GetPublish(id string) Publish {
 	out.raw.Links["self"] = url
 	out.raw.Links["commit"] = url + "/commit"
 
-	return out
+	// Verify that the publish ID is valid before uploading blobs.
+	empty := struct{}{}
+	if err := c.doJSONRequest(ctx, "GET", url, nil, &empty); err != nil {
+		return out, err
```

Comment:
In the case where an error occurs, can we please make it return 'nil' instead of 'out'?

The original design is that it would never fail, so it would always have an object to return.  However, since there is now a failure case, it doesn't seem a good idea to still return an object in the case where it failed.  By returning an object we run the risk that some code will ignore the 'err' and still try to use the publish object.

(In fact, I think client_publish_test.go is currently making that mistake, which would be caught if you made it return nil.)